### PR TITLE
fix old style function definition

### PR DESCRIPTION
**Pull request by ElyesH**

### CODE REVIEW

File: usr.sbin/cron/cron/externs.h [hunk 1]
```diff
@@ -49,14 +49,14 @@ extern char *tzname[2];
...
+extern	unsigned	sleep(void);
+extern	struct tm	*localtime(void);
+extern	struct passwd	*getpwnam(void);
 extern	int		errno;
-extern	void		perror(), exit(), free();
-extern	char		*getenv(), *strcpy(), *strchr(), *strtok();
-extern	void		*malloc(), *realloc();
+extern	void		perror(void), exit(void), free(void);
+extern	char		*getenv(void), *strcpy(void), *strchr(void), *strtok(void);
+extern	void		*malloc(void), *realloc(void);
```
Comment (igalic) [inline usr.sbin/cron/cron/externs.h:59]: this feels confusing.
it makes sense that it's possible, but I've never seen it like this.

**Review by igalic — COMMENTED** [review]
👀

File: usr.sbin/cron/cron/externs.h (hunk 1 above)
Comment (ElyesH) [reply usr.sbin/cron/cron/externs.h:59]: Also, "extern"  is used implicitly for functions declarations.

File: usr.sbin/cron/cron/database.c [hunk 2]
```diff
@@ -237,7 +237,7 @@ find_user(db, name)
 	cron_db	*db;
 	char	*name;
 {
-	char	*env_get();
+	char	*env_get(void);
```
Comment (bsdimp) [inline usr.sbin/cron/cron/database.c:240]: This is wrong too. It looks like the line can just be deleted.

File: usr.sbin/cron/cron/externs.h (hunk 1 above)
Comment (bsdimp) [reply usr.sbin/cron/cron/externs.h:59]: I'd delete all of these, and then include the proper .h file to get them instead...

File: usr.sbin/cron/cron/popen.c [hunk 3]
```diff
@@ -74,7 +74,7 @@ cron_popen(program, type, e, pidptr)
 	char **pop, *vv[2];
 	int gargc;
 	char *gargv[1000];
-	extern char **glob(), **copyblk();
+	extern char **glob(void), **copyblk(void);
```
Comment (bsdimp) [inline usr.sbin/cron/cron/popen.c:77]: These are also wrong. These take args.
I'm unsure where it's snagging these from. glob looks to come from libc, not sure copyblk.
glob's .h file is glob.h.

File: usr.sbin/keyserv/keyserv.c [hunk 4]
```diff
@@ -83,9 +83,9 @@ static int debugging = 1;
 static int debugging = 0;
 #endif
 
-static void keyprogram();
+static void keyprogram(void);
```
Comment (bsdimp) [inline usr.sbin/keyserv/keyserv.c:86]: This is wrong, it takes 2 args.

File: usr.sbin/keyserv/keyserv.c [hunk 5]
```diff
@@ -83,9 +83,9 @@ static int debugging = 1;
...
 static des_block masterkey;
-char *getenv();
+char *getenv(void);
```
Comment (bsdimp) [inline usr.sbin/keyserv/keyserv.c:88]: We should delete this and get the definition from <stdlib.h>

File: usr.sbin/ypbind/ypbind.c [hunk 6]
```diff
@@ -91,9 +91,9 @@ struct _dom_binding {
 #define WRITEFD ypdb->dom_pipe_fds[1]
 #define BROADFD broad_domain->dom_pipe_fds[1]
 
-extern bool_t xdr_domainname(), xdr_ypbind_resp();
-extern bool_t xdr_ypreq_key(), xdr_ypresp_val();
-extern bool_t xdr_ypbind_setdom();
+extern bool_t xdr_domainname(void), xdr_ypbind_resp(void);
+extern bool_t xdr_ypreq_key(void), xdr_ypresp_val(void);
+extern bool_t xdr_ypbind_setdom(void);
```
Comment (bsdimp) [inline usr.sbin/ypbind/ypbind.c:96]: These all take args, not sure where a .h for them is off the top of my head.

File: usr.sbin/ypserv/common/yplib_host.c [hunk 7]
```diff
@@ -56,8 +56,8 @@ __FBSDID("$FreeBSD$");
 
 extern int (*ypresp_allfn)(u_long, char *, int, char *, int, void *);
 extern void *ypresp_data;
-extern bool_t xdr_ypreq_key(), xdr_ypresp_val();
-extern bool_t xdr_ypresp_all_seq();
+extern bool_t xdr_ypreq_key(void), xdr_ypresp_val(void);
+extern bool_t xdr_ypresp_all_seq(void);
```
Comment (bsdimp) [inline usr.sbin/ypserv/common/yplib_host.c:60]: These all take args.

**Comment by bsdimp** [comment]
This isn't ready, I'm surprised it even compiles in spots.

**Comment by bsdimp** [comment]
I'm going to close this now. Please resubmit when it's ready.

**Pull request closed without merging**